catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveTableOperations.java

```diff
@@ -215,8 +222,107 @@ private org.apache.hadoop.hive.metastore.api.Partition toHivePartition(
   }
 
   @Override
-  public boolean dropPartition(String partitionName) {
-    throw new UnsupportedOperationException();
+  public boolean dropPartition(String partitionName, boolean ifExists)
+      throws NoSuchPartitionException {
+    try {
+      Table hiveTable = table.clientPool().run(c -> c.getTable(table.schemaName(), table.name()));
+      // Get partitions that need to drop
+      // If the partition has child partition, then drop all the child partitions of the parent partition cascade
+      // If the partition is a completed partition, then just drop one partition
```

Comment:
`If the partition is a completed partition` -> `If the partition has no subpartitions`

Reply:
done